src/flag_gems/ops/softmax.py

```diff
@@ -180,15 +180,18 @@ def softmax_kernel_inner(
             n_offsets = start_n + tl.arange(0, TILE_N)
             inp = tl.load(input_ptr + n_offsets)
             m_new = tl.maximum(m, inp)
-            z = z * tl.exp(m - m_new) + tl.exp(inp - m_new)
+            # it is possible that there are -inf's in the input
+            all_neg_inf = m_new == float("-inf")
```

Comment:
I'm a little bit curious whether float(-inf) is casted to fp64 if inp is fp64

Reply:
Yes， it is.

Check `binary_op_type_checking_impl` and `computation_type_impl` for more details.